src/payment/data/sagas.js

```diff
@@ -47,64 +43,46 @@ export function* handleFetchBasket() {
   }
 }
 
+function* isBasketProcessing() {
+  return yield select(state => state.payment.basket.isBasketProcessing);
+}
+
+export function* handleFetchBasket() {
+  yield call(catchBasketError, function* processFetchBasket() {
```

Comment:
These are much shorter now!

Reply:
Reads weird to me but definitely does the job.